src/main/java/net/snowflake/ingest/connection/SecurityManager.java

```diff
@@ -176,4 +198,82 @@ String getToken()
     return token.get();
   }
 
+  /**
+   * Hashes input bytes using SHA-256.
+   *
+   * @param input  input bytes
+   * @param offset offset into the output buffer to begin storing the digest
+   * @param len    number of bytes within buf allotted for the digest
+   * @return SHA-256 hash
+   */
+  public static byte[] sha256Hash(byte[] input, int offset, int len)
```

Comment:
There is no reason for all these new methods in this file to be public. Please make them private.